lib/Sema/TypeCheckStorage.cpp

```diff
@@ -1416,13 +1416,24 @@ synthesizeObservedSetterBody(AccessorDecl *Set, TargetImpl target,
   SmallVector<ASTNode, 6> SetterBody;
 
   auto callObserver = [&](AccessorDecl *observer, VarDecl *arg) {
+    // If we're not loading the old value, then remove the oldValue parameter
+    // and re-compute the type.
+    if (!arg) {
+      observer->setParameters(ParameterList::createEmpty(Ctx));
+      observer->computeType();
```

Comment:
I would still prefer if the decision to add (or not) the oldValue happened earlier, when the observer was created. Changing the interface type of a declaration will soon be banned.

Reply:
You mean in the parser?